sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/externalrules/CxxExternalRulesSensor.java

```diff
@@ -59,11 +72,30 @@ protected String reportPathKey() {
   public void describe(SensorDescriptor descriptor) {
     descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxExternalRulesSensor");
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
   @Override
-  public void processReport(final SensorContext context, File report) throws XMLStreamException {
+  public void execute(SensorContext context) {
+    try {
+      transformFiles();
+    } catch (Exception e) {
```

Comment:
Up to now we have two modes: strict and tolerant
* strict: break whole analysis after an error
* tolerant: ignore and continue

This is hidden behind CxxUtils.validateRecovery.

In tolerant mode:
* should continue if one of the XSLT transformation fails and there are more source files in the comma separated list
* should continue with next XSLT transformation if one before was failing